journal/src/main/java/io/zeebe/journal/file/MappedJournalSegmentWriter.java

```diff
@@ -205,6 +210,14 @@ private void reset(final long index) {
       }
     } catch (final BufferUnderflowException e) {
       // Reached end of the segment
+    } catch (final CorruptedLogException e) {
+      // if at startup, assume record was partially written due to crash and mark as ignored
+      if (!startup) {
+        throw e;
+      }
+
+      buffer.reset();
```

Comment:
I think this does not work as expected because there is another `buffer.mark()` in the JournalReaderUtil. This is one problematic thing when using ByteBuffers and passing it around. We have already discussed that we want to remove ByteBuffers and probably use DirectBuffers everywhere. But until then we have to use `mark()` and `reset()` carefully. I would suggest to record the position instead of mark and use `buffer.position(markedPosition)` instead of reset.